requirements-dev.txt

```diff
@@ -1,340 +1,41 @@
-#
-# This file is autogenerated by pip-compile with Python 3.11
-# by the following command:
-#
-#    pip-compile --strip-extras k8s/cli/requirements-dev.in
-#
-anyio==4.4.0
```

Comment:
Some of these requirements have been removed because they are part of requirements.txt

The requirements-dev.in file was incorrectly adding the regular reauirements.in file [here](https://github.com/getsentry/sentry-infra-tools/pull/23/files#diff-5de99c387a95507082f5225072501f952e3dc9a6c839192d0ccd04985005596cL1) which would lead to duplicate requirement entries.